activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -30,12 +30,8 @@ class SQLite3Adapter < AbstractAdapter
       class << self
         def new_client(config)
           ::SQLite3::Database.new(config[:database].to_s, config)
-        rescue Errno::ENOENT => error
-          if error.message.include?("No such file or directory")
```

Comment:
This feels like we're conflating two possible error scenarios, however:

* When SQLite can't connect because the database doesn't exist
* When there is another error which results in unable to read the db, think permission

These are different error states, and by addressing this here we are impacting more than just the `db:drop` command.

Is it possible to set this flag _before_ executing the rake command?

I don't think there is any point in wrapping `SQLite3::CantOpenException` in `AR::NoDatabaseError`.